taichi/aot/module_loader.cpp

```diff
@@ -1,18 +1,34 @@
 #include "taichi/aot/module_loader.h"
 
+#ifdef TI_WITH_VULKAN
```

Comment:
Just hide these in each backend's impl. It's fine that `xx_backend::make_aot_module()` is **declared** in the header

Reply:
Just to confirm, you mean to add these declarations in `module_loader.h` is fine? 
```
namespace vulkan {
std::unique_ptr<aot::Module> make_aot_module(std::any mod_params);
}  // namespace vulkan 
 
namespace metal {
std::unique_ptr<aot::Module> make_aot_module(std::any mod_params);
}  // namespace metal 
```

Reply:
changed